src/main/java/com/pinterest/secor/writer/MessageWriter.java

```diff
@@ -75,14 +75,15 @@ public MessageWriter(SecorConfig config, OffsetTracker offsetTracker,
         mGeneration = mConfig.getGeneration();
     }
 
-    public void adjustOffset(Message message) throws IOException {
+    public void adjustOffset(Message message, boolean isLegacyConsumer) throws IOException {
         TopicPartition topicPartition = new TopicPartition(message.getTopic(),
                                                            message.getKafkaPartition());
         long lastSeenOffset = mOffsetTracker.getLastSeenOffset(topicPartition);
-        if (message.getOffset() != lastSeenOffset + 1) {
+        //this validation logic is for duplicates removing as no rebalancing callbacks is incompatible with LegacyKafkaMessageIterator
+        if (isLegacyConsumer && message.getOffset() < lastSeenOffset + 1) {
```

Comment:
Should you use the old logic: 'message.getOffset() != lastSeenOffset +1' here?

Reply:
sure. will do an update.